process.py

```diff
@@ -863,6 +883,54 @@ def git_pull(gitdir):
     repo = git.Repo(gitdir)
     repo.remotes.origin.pull()
     return git_repo_state(repo)
+
+#
+#    GitLab
+#
+
+def gitlab_src_from_fork(repo_fork):
+    if project.forked_from_project:
+        source_project_id = project.forked_from_project['id']
+        source_project = gl.projects.get(source_project_id)
+        print(f"Source Project: {source_project.web_url}")
+        return source_project
+    else:
+        print("The project is not a fork.")
+        return None
+
+def gitlab_create_mr(repo_fork, repo_target, bugnumber, branch='main'):
+    arguments = {
+        'source_branch': branch,
+        'target_branch': branch,
+        'target_project_id' : repo_target.id,
+        'assignee_id' : GITLAB.user.id,
+        'title': (bug_summary_short % year),
+        'description' : ("Resolves: %s\n\n" % bugnumber),
+    }
+
+    mr = repo_fork.mergerequests.create(arguments)
+
+    return mr
+
+def gitlab_find_mr(upstream_project, source_branch, source_project_id):
+    mrs = upstream_project.mergerequests.list()
+    for mr in mrs:
+        if mr.source_branch == source_branch and \
+           mr.source_project_id == source_project_id and \
+           mr.title == (bug_summary_short % year) and \
+           mr.description == ("Resolves: %s\n\n" % bugnumber):
```

Comment:
It might be better to do regulare expression test there  for Resolves:.*bugnumber.* so that we can handle merge requests that are manually created. (When steps fail in the automation, we often have to resort to manual intervension. As much as we can recognized manually created entries the better).